### PR TITLE
Use open telemetry

### LINKED ISSUE #196: Consolidate to opentelemetry
**Is your feature request related to a problem? Please describe.**
Opentracing requires vendor-specific clients.

**Describe the solution you'd like**
Use opentelemetry with the collector, which makes COS vendor-agnostic.

**Describe alternatives you've considered**
bundling all clients into COS

**Additional context**
this was raised by the Glean team.

### PR DESCRIPTION
**Pull request by klvmungai**
fixes #196

### CODE REVIEW

File: code/service/src/main/scala/com/namely/chiefofstate/telemetry/TelemetryTools.scala [hunk 1]
```diff
@@ -6,67 +6,93 @@
...
+        .setMetricExporter(exporter)
+      val reader = readerBuilder.build
+      sys.addShutdownHook {
+        reader.shutdown()
+      }
     }
+  }
+
+  def configureProvider(config: Config, resource: Resource): Option[SdkTracerProvider] = {
+    Option(config.getString("chiefofstate.otlp.endpoint"))
```
Comment (Tochemey) [inline code/service/src/main/scala/com/namely/chiefofstate/telemetry/TelemetryTools.scala:65]: Let us use the CosConfig to set this. That will help us know that all config are centralized.

File: code/service/src/main/scala/com/namely/chiefofstate/telemetry/TelemetryTools.scala [hunk 2]
```diff
@@ -6,67 +6,93 @@
...
 
-  /**
-   * stop all telemetry tools
-   */
-  def stop(): Unit = {
-    logger.debug("stopping telemetry tools")
-    this.prometheusServer.stop()
+  def configureResource(config: Config): Resource = {
+    val resourceAttributes = Attributes.builder()
+    resourceAttributes.put(ResourceAttributes.SERVICE_NAME, config.getString("chiefofstate.service-name"))
```
Comment (Tochemey) [inline code/service/src/main/scala/com/namely/chiefofstate/telemetry/TelemetryTools.scala:89]: Same here

File: code/service/src/main/scala/com/namely/chiefofstate/telemetry/TelemetryTools.scala [hunk 3]
```diff
@@ -6,67 +6,93 @@
...
+
+    Option(config.getString("chiefofstate.namespace"))
```
Comment (Tochemey) [inline code/service/src/main/scala/com/namely/chiefofstate/telemetry/TelemetryTools.scala:91]: Same here

**Review by Tochemey — COMMENTED** [review]
Looks good.

**Review by Tochemey — APPROVED** [review]

File: code/service/src/main/scala/com/namely/chiefofstate/AggregateRoot.scala [hunk 4]
```diff
@@ -101,21 +99,23 @@ object AggregateRoot {
...
+    val ctx = TracingHelpers.getParentSpanContext(Context.current(), headers)
 
-    val span: Span = OpentracingHelpers
-      .getChildSpanBuilder(tracer, headers, "AggregateRoot.handleCommand")
-      .withTag(Tags.COMPONENT.getKey(), this.getClass().getName)
-      .start()
+    val span: Span = GlobalOpenTelemetry
+      .getTracer(getClass.getPackage.getName)
+      .spanBuilder("AggregateRoot.handleCommand")
+      .setAttribute("component", this.getClass.getName)
```
Comment (zenyui) [inline code/service/src/main/scala/com/namely/chiefofstate/AggregateRoot.scala:109]: Are there constants for this?

File: code/service/src/main/scala/com/namely/chiefofstate/telemetry/PropagatorConfiguration.scala [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+
+import com.namely.chiefofstate.config.TelemetryConfig
+import io.opentelemetry.api.baggage.propagation.W3CBaggagePropagator
+import io.opentelemetry.api.trace.propagation.W3CTraceContextPropagator
+import io.opentelemetry.context.propagation.{ContextPropagators, TextMapPropagator}
+import io.opentelemetry.extension.trace.propagation.{B3Propagator, JaegerPropagator, OtTracePropagator}
+
+import scala.jdk.CollectionConverters._
+
+object PropagatorConfiguration {
```
Comment (zenyui) [inline code/service/src/main/scala/com/namely/chiefofstate/telemetry/PropagatorConfiguration.scala:17]: What is this?

**Review by zenyui — APPROVED** [review]
Really nice work. Just add docstrings :)

**Review by Tochemey — APPROVED** [review]

**Review by zenyui — APPROVED** [review]

**Pull request merged**